src/js/angular/similarity/controllers/similarity-list.controller.js

```diff
@@ -360,26 +372,31 @@ function SimilarityCtrl($scope, $interval, toastr, $repositories, $licenseServic
         if ($scope.loadSimilarityIndexesTimer) {
             $interval.cancel($scope.loadSimilarityIndexesTimer);
         }
-    });
+    };
+
+    // Deregister the watcher when the scope/directive is destroyed
+    $scope.$on('$destroy', destroyHandler);
 
     const removeAllListeners = () => {
         subscriptions.forEach((subscription) => subscription());
     };
 
     subscriptions.push($scope.$on('autocompleteStatus', checkAutocompleteStatus));
 
-    subscriptions.push($scope.$watch('searchType', function () {
+    const setEmptyHandler = () => {
         $scope.empty = true;
-    }));
+    };
 
-    // Wait until the active repository object is set, otherwise "canWriteActiveRepo()" may return a wrong result and the "ontotext-yasgui"
-    // readOnly configuration may be incorrect.
-    const repoIsInitialized = $scope.$watch(function () {
-        return $scope.getActiveRepositoryObject();
-    }, function (activeRepo) {
+    subscriptions.push($scope.$watch('searchType', setEmptyHandler));
+
+    const getActiveRepositoryObjectHandler = (activeRepo) => {
         if (activeRepo) {
             init();
             repoIsInitialized();
         }
-    });
+    };
+
+    // Wait until the active repository object is set, otherwise "canWriteActiveRepo()" may return a wrong result and the "ontotext-yasgui"
+    // readOnly configuration may be incorrect.
+    const repoIsInitialized = $scope.$watch($scope.getActiveRepositoryObject, getActiveRepositoryObjectHandler);
```

Comment:
And same here. Rename it to `activeRepositoryChangeHandler`

Reply:
Done

Reply:
Done